### PR TITLE
fix: some end points are not routed through the proxy server 4.6

### PR DESCRIPTION
**Pull request by AndroidBob**
Cherry pick from the original PR: 
- #2723

---- 

 ⚠️ Conflicts during cherry-pick:
app/src/main/res/values-de/strings.xml

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like 
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
some end points are not routed through the proxy server mainly the api version

### Solutions

refactore the code so every thing is under the unauthenticated container and make sure is does have the proxy info passed correctly

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

run the app connet to wire server via socks proxy (message me if you need the setup) and in the network inspection only proxy link is used

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. .

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3412](https://10.10.124.134/job/AR-build-pipeline/3412/) **failed**.

**Comment by AndroidBob** [comment]
Build [3654](https://10.10.124.134/job/AR-build-pipeline/3654/) **failed**.

**Comment by AndroidBob** [comment]
Build [3655](https://10.10.124.134/job/AR-build-pipeline/3655/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9187601)(9187601)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9187601)(9187601)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt [hunk 1]
```diff
@@ -134,7 +136,9 @@ class LoginSSOViewModel @Inject constructor(
             val defaultAuthScope: AuthenticationScope =
                 coreLogic.versionedAuthenticationScope(
                     authServerConfigProvider.defaultServerLinks()
-                )().let {
+                    // domain lockup does not support proxy
+                    // TODO: add proxy support
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt:140]: For later?

**Review by mchenani — APPROVED** [review]
Lgtm, added a question.

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt (hunk 1 above)
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt:140]: domain lookup happen on wire servers and the result of it is a new on prem server config that the user can login with.
for the time being proxy is not supported with that feature but the result server config can have a vproxy and it works just fine

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**